test/generators/java/__snapshots__/JavaGenerator.spec.ts.snap

```diff
@@ -298,6 +308,178 @@ Array [
 ]
 `;
 
+exports[`JavaGenerator oneOf/discriminator with jackson preset handle setting title with const 1`] = `
+Array [
+  "@JsonTypeInfo(use=JsonTypeInfo.Id.NAME, include=JsonTypeInfo.As.PROPERTY, property=\\"type\\")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = Dog.class, name = \\"Dog\\"),
+  @JsonSubTypes.Type(value = Cat.class, name = \\"Cat\\")
+})
+/**
+ * Pet represents a union of types: Dog, Cat
+ */
+public interface Pet {
+  
+}",
+  "public class Dog implements Pet {
+  @NotNull
+  @JsonProperty(\\"type\\")
+  private final DogType type = DogType.DOG;
+  private Map<String, Object> additionalProperties;
+
+  public DogType getType() { return this.type; }
```

Comment:
nit: this should have an `@Override` annotation to be consistent with other generated code bellow. Not functionally required, and we should consider not doing this as the implementation is likely to be a bit fiddly.

Reply:
...oh, this isn't exposed via the interface, so it isn't overridden.

I would expect this to be an enum `PetType` then exposed via a getter in the interface. This is likely to interact interestingly with the const code.

Reply:
the test "JavaGenerator oneOf/discriminator with jackson preset handle allOf with const in CloudEvent type" uses a common type for the disriminator, so I think we have a mild conflict between the `union` and `const` behavour

Reply:
Idiomatically in java you would make a constant by declaring the underlying field as `final`

Reply:
OK, after discussion, I'm happy that this is something the user is in full control of. It is possible to write a schema that breaks things in the java code gen, but the users isn't guided down that path.